### PR TITLE
module1

### PR DESCRIPTION
**Pull request by ivdanchuk**

### CODE REVIEW

File: module1/src/components/Counter.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React, { Component, ComponentClass } from "react";
+
+class Counter extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      counter: this.props.initValue || 0,
```
Comment (StanislavDereviankoEPAM) [inline module1/src/components/Counter.js]: this.props.initValue - what is this? Where did you use it?

File: module1/src/components/Counter.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    };
+  }
+  increment = () => {
+    this.setState((prevState) => ({ counter: this.state.counter + 1 }));
```
Comment (StanislavDereviankoEPAM) [inline module1/src/components/Counter.js]: Why do you need prevState here?

File: module1/src/components/GenreSelect.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class GenreSelect extends React.Component {
+  render() {
+    const { genres, selectedGenre, onSelect } = this.props;
+    return (
+      <div className="genre-select">
+        {genres.map((genre) => (
+          <button
```
Comment (StanislavDereviankoEPAM) [inline module1/src/components/GenreSelect.js:10]: Look into design prototype, it's not a button

File: module1/src/components/GenreSelect.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            key={genre}
+            className={`genre-select-button ${
```
Comment (StanislavDereviankoEPAM) [inline module1/src/components/GenreSelect.js]: You should use styles to make it work, less, sass, etc..

File: module1/src/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import ReactDOM from "react-dom/client";
+// import "./index.css";
```
Comment (StanislavDereviankoEPAM) [inline module1/src/index.js]: remove unused code

File: module1/src/App.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { Component } from "react";
+import "./css/App.css";
+import Counter from "./components/Counter";
+import SearchForm from "./components/SearchForm";
+import GenreSelect from "./components/GenreSelect";
+
+class App extends Component {
+  handleSearch = (query) => {
```
Comment (StanislavDereviankoEPAM) [inline module1/src/App.js]: Remove console.log, we don't need it

File: module1/src/App.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    console.log("Searching:", query);
+  };
+
+  handleSelect = (genre) => {
+    console.log("Selected genre:", genre);
```
Comment (StanislavDereviankoEPAM) [inline module1/src/App.js]: onSelect you should change style for genre button, console.log is  incorrect here

File: module1/src/components/Counter.js (hunk 1 above)
Comment (ivdanchuk) [reply module1/src/components/Counter.js]: <Counter initValue={10}></Counter>

File: module1/src/components/Counter.js (hunk 2 above)
Comment (ivdanchuk) [reply module1/src/components/Counter.js]: fixed

File: module1/src/App.js (hunk 6 above)
Comment (ivdanchuk) [reply module1/src/App.js]: How can you see working callback without console.log ?

Comment (StanislavDereviankoEPAM) [reply module1/src/App.js]: I can read the code =)

File: module1/src/components/GenreSelect.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+
+class GenreSelect extends React.Component {
+  render() {
+    const { genres, selectedGenre, onSelect } = this.props;
+    return (
+      // <div className="genre-select">
```
Comment (StanislavDereviankoEPAM) [inline module1/src/components/GenreSelect.js]: All commented code should be removed

File: module1/src/App.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        <div className="header">
+          <SearchForm initialQuery="" onSearch={this.handleSearch} />
+        </div>
+        <GenreSelect
+          genres={this.state.genres}
+          selectedGenre={this.state.selectedGenre}
+          onSelect={this.handleSelect}
+        />
+        <div>
+          <Counter initValue={10}></Counter>
```
Comment (StanislavDereviankoEPAM) [inline module1/src/App.js:32]: Component in render() without children could be used as self-closed: <Counter initValue={10} />

File: module1/src/App.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import { Component } from "react";
+import "./css/App.scss";
+import SearchForm from "./components/SearchForm";
+import GenreSelect from "./components/GenreSelect";
+import Counter from "./components/Counter";
+
+class App extends Component {
+  state = {
+    genres: ["ALL", "Crime", "Comedy", "Drama", "Horror"],
```
Comment (StanislavDereviankoEPAM) [inline module1/src/App.js]: Such type of constants could be moved to constants folder and then be imported

File: module1/src/css/App.scss [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+.search-container button {
+  background-color: #f65261;
+  border: none;
+  color: white;
+}
+.movie-genre {
+  display: flex;
+  //justify-content: space-around;
```
Comment (StanislavDereviankoEPAM) [inline module1/src/css/App.scss]: remove commented code

File: module1/src/css/App.scss [hunk 12]
```diff
@@ -0,0 +1,47 @@
+.header {
```
Comment (StanislavDereviankoEPAM) [inline module1/src/css/App.scss]: don't mix scss and css, rename this folder as: styles
also this file contains all styles for all components, create separated folders for each component with styles, index.js etc..

**Pull request merged**